engine/api/src/main/java/io/deephaven/engine/table/ColumnDefinition.java

```diff
@@ -504,6 +504,28 @@ public void describeDifferences(@NotNull List<String> differences, @NotNull fina
         }
     }
 
+    /**
+     * Checks if {@link #getDataType() dataType} can be cast to {@code destDataType}. If not, this throws a
+     * {@link ClassCastException}.
+     *
+     * @param destDataType the destination data type
+     */
+    public final void checkCastTo(Class<?> destDataType) {
```

Comment:
Usually we just have a `cast` method that returns the type we like and (maybe) validates internally.

Reply:
I think it's a little bit different w/ `ColumnDefinition`. In other cases cases where `cast` is used, it is typically done to match types b/c the consumer wants _to get data out_. In this case, `ColumnDefinition` itself doesn't hold any data, it is just parameterized to support explicit typing around `Class <TYPE> getDataType()`.

I don't see a practical use for wanting to actually cast a definition. Furthermore, it breaks the contract:

```java
ColumnDefinition<Integer> intColDef = ...;
ColumnDefinition<Number> numColDef = intColDef.cast(Number.class);

Class<Number> numberClass = numColDef.getDataType();
// This will fail
assertEquals(Number.class, numberClass);
```

We do play loose and fast with this in other places; `ColumnSource` suffers from the same `getType` problem - although again, `ColumnSource#cast` is usually used to get _data_ out (not b/c somebody wants `getType`).

In some more recent code, I've tried to "do the right thing"; `io.deephaven.functions.ToObjectFunction#cast`